src/process-events.c

```diff
@@ -173,5 +171,41 @@ int BPF_KPROBE_SYSCALL(kprobe__sys_exit_group, int status)
     return 0;
 }
 
+SEC("kprobe/sys_execveat")
+int BPF_KPROBE_SYSCALL(kprobe__sys_execveat,
+                       int fd, const char __user *filename,
+                       const char __user *const __user *argv,
+                       const char __user *const __user *envp,
+                       int flags)
+{
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    u32 pid = pid_tgid >> 32;
```

Comment:
I hate that `pid` means something different in the kernel than userspace.  I also hate that the names for these dual-32-bit helpers seem reversed from the layout of the value they return.

IIRC `pid_tgid >> 32` actualy gives us `tgid` which is what userspace considers the `pid`, but I wonder if we might call it some third thing instead so I don't end up second-guessing myself every time I read through these.  I can't think of anything equally short and clear though, so I'll leave it to your judgement.

Reply:
I agree it is pretty confusing. Right now what I've been doing is that as soon as the field is "translated" by doing the bitshift or mask I start naming them in terms of user-space. So in `u32 pid = pid_tgid >> 32`, `pid` is the user space process id (kernel space thread group id). In other spots where we do `u32 tid = pid_tgid & 0xFFFFFFFF`, `tid` is the user space threads id (kernel space process id).